draft-documents/encrypted-data-vaults.md

```diff
@@ -79,79 +69,172 @@ one or more device(s) they control, and data is stored locally to these devices.
 IPFS is peer-to-peer, so end users only install the read/write client, and data
 is stored across a public network.
 
-NextCloud uses WebDAV to allow client applications to read, write, and search
-data on the server's filesystem using a directory structure, and OCP for
-authentication. End-to-end encryption is currently in beta and can be enabled by
-users, but is not on by default. Spreading data across multiple instances for
-scalability is a [commercial enterprise
-offering](https://nextcloud.com/globalscale/). Different NextCloud servers do
-not talk to each other directly, but can do via applications installed by the
-user.
-
+Identity Hubs are responsible for additional things beyond just data storage,
+for example management of the end user's profile; transmission of human- or
+machine-readable messages through the Actions interface; pointers to external
+services.
+
+### Encryption policies
+
+An important consideration of encrypted data stores is which components of the
+architecture have access to the (unencrypted) data, or who controls the private
+keys. There are roughly three approaches: storage-side encryption, client-side
+(edge) encryption, and gateway-side encryption (which is a hybrid of the
+previous two).
+
+Any data storage systems which let the user store arbitrary data support
+client-side encryption at the most basic level. That is, they let the user
+encrypt data themselves, and then store it. This doesn’t mean these systems are
+optimised for encrypted data however (eg. querying and sharing may be difficult),
+or that they won’t leak data in other ways (eg. through unencrypted metadata).
+Solid, NextCloud, Identity Hubs and IPFS take this approach.
```

Comment:
Maybe we can say something like:

This doesn’t mean these systems are optimised for encrypted data however. Querying and access control for encrypted data may be difficult (as is the case for Solid, NextCloud and IPFS). And querying un-encrypted metadata (as in the current design for Identity Hubs) may leak confidential information in other ways.

Reply:
Yes minus the last sentence because metadata and querying are covered more later